kaska/s2_observations.py

```diff
@@ -224,43 +263,46 @@ def read_granule(self, timestep):
             rho = reproject_data(
                 str(original_s2_file), target_img=self.state_mask
             ).ReadAsArray()
-            
+
             rho_surface.append(rho)
             original_s2_file = current_folder / (
                 f"{fname_prefix:s}" + f"{the_band:s}_sur_unc.tif"
             )
-            LOG.debug(f"Uncertainty file {str(original_s2_file):s}")
-            unc = reproject_data(
-                str(original_s2_file), target_img=self.state_mask
-            ).ReadAsArray()
-            rho_unc.append(unc)
+            # LOG.debug(f"Uncertainty file {str(original_s2_file):s}")
+            # unc = reproject_data(
+            #    str(original_s2_file), target_img=self.state_mask
+            # ).ReadAsArray()
+            rho_unc.append(np.ones_like(rho) * 0.005)
         # For reference...
-        #bands = ['B01', 'B02', 'B03', 'B04', 'B05', 'B06', 'B07', 'B08',
+        # bands = ['B01', 'B02', 'B03', 'B04', 'B05', 'B06', 'B07', 'B08',
         #         'B8A', 'B09', 'B10','B11', 'B12']
         # b_ind = np.array([1, 2, 3, 4, 5, 6, 7, 8])
-        
-        rho_surface = np.array(rho_surface)
-        
-        
-        mask1 = np.all(rho_surface[[1, 2, 3, 4, 5, 6, 7, 8, ]] > 0,
-                       axis=0) & (~mask)
-        mask = ~mask1
-        rho_surface = rho_surface / 10000.0
-        
-        #mask = np.all(rho_surface > 0, axis=0) & (~mask)
 
-        rho_unc = np.array(rho_unc) / 10000.0
-        rho_unc[:, mask] = np.nan
-        # Average uncertainty over the image
-        rho_unc = np.nanmean(rho_unc, axis=(1, 2))
-        rho_surface[:, mask1] = np.nan
+        rho_surface = np.array(rho_surface)
+        # Now, ensure all surface reflectance pixels have values above
+        # 0 & aren't cloudy.
+        # So valid pixels if all refl > 0 AND mask is True
+        sel_bands = np.array([1, 2, 3, 4, 5, 6, 7, 8])
```

Comment:
```suggestion
        sel_bands = np.arange(1,9)
```

Reply:
The bands are consecutive now, there's no guarantee that they have to remain so.

Reply:
I think add a comment in the code about that, @timspainUCL. So it doesn't accidentally get changed in future.